tests/test_parser.py

```diff
@@ -781,6 +781,82 @@ def function() -> int:
         assert parser.raw_statements == {1, 3, 4, 5, 6, 8, 9}
         assert parser.statements == {1, 8, 9}
 
+    def test_multiline_exclusion_single_line(self) -> None:
+        regex = r"print\('.*'\)"
+        parser = self.parse_text("""\
+            def foo():
+                print('Hello, world!')
+        """, regex)
+        assert parser.lines_matching(regex) == {2}
+        assert parser.raw_statements == {1, 2}
+        assert parser.statements == {1}
+
+    def test_multiline_exclusion_suite(self) -> None:
+        regex = r"if T:\n\s+print\('Hello, world!'\)"
+        parser = self.parse_text("""\
+            def foo():
+                if T:
+                    print('Hello, world!')
+                    print('This is a multiline regex test.')
+        """, regex)
+        assert parser.lines_matching(regex) == {2, 3}
+        assert parser.raw_statements == {1, 2, 3, 4}
+        assert parser.statements == {1}
+
+    def test_multiline_exclusion_no_match(self) -> None:
+        regex = r"nonexistent"
+        parser = self.parse_text("""\
+            def foo():
+                print('Hello, world!')
+        """, regex)
+        assert parser.lines_matching(regex) == set()
+        assert parser.raw_statements == {1, 2}
+        assert parser.statements == {1, 2}
+
+    def test_multiline_exclusion_no_source(self) -> None:
+        regex = r"anything"
+        parser = PythonParser(text="", filename="dummy.py", exclude=regex)
+        assert parser.lines_matching(regex) == set()
+        assert parser.raw_statements == set()
+        assert parser.statements == set()
+
+    def test_multiline_exclusion_multiple_matches(self) -> None:
+        regex = r"print\('.*'\)"
```

Comment:
This isn't a multi-line regex?

Reply:
Right, that was just a sanity check from when I was sketching this that slipped through. I'll rewrite the test so it covers multiple multi-line matches.